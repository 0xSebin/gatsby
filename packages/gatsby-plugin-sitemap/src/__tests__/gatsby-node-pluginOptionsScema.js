/* eslint-disable filenames/match-regex */
process.env.GATSBY_EXPERIMENTAL_PLUGIN_OPTION_VALIDATION = true

jest.mock(`sitemap`, () => {
  return {
    simpleSitemapAndIndex: jest.fn(),
  }
})

const sitemap = require(`sitemap`)

const { validateOptionsSchema, Joi } = require(`gatsby-plugin-utils`)
const { onPostBuild, pluginOptionsSchema } = require(`../gatsby-node`)

const pathPrefix = ``

const reporter = {
  verbose: jest.fn(),
  panic: jest.fn(),
}

const graphql = jest.fn()
graphql.mockResolvedValue({
  data: {
    site: {
      siteMetadata: {
        siteUrl: `http://dummy.url`,
      },
    },
    allSitePage: {
      nodes: [
        {
          path: `/page-1`,
        },
        {
          path: `/page-2`,
        },
      ],
    },
  },
})

beforeEach(() => {
  global.__PATH_PREFIX__ = ``

  sitemap.simpleSitemapAndIndex.mockReset()
})

describe(`gatsby-plugin-sitemap Node API with Expermintal plugin option Validation`, () => {
  it(`should succeed when using \`pluginOptionsSchema\` API`, async () => {
    const optionSchema = pluginOptionsSchema({ Joi })

    const pluginOptions = await validateOptionsSchema(optionSchema, {})

    expect(pluginOptions).toMatchSnapshot()

    await onPostBuild({ graphql, pathPrefix, reporter }, pluginOptions)
  })
})
