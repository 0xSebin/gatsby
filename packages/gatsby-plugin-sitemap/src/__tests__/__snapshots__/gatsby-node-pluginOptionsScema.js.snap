// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-sitemap Node API with Expermintal plugin option Validation should succeed when using \`pluginOptionsSchema\` API 1`] = `
Object {
  "createLinkInHead": true,
  "entryLimit": 45000,
  "excludes": Array [],
  "filterPages": [Function],
  "output": "/sitemap",
  "query": "{ site { siteMetadata { siteUrl } } allSitePage { nodes { path } } }",
  "resolvePagePath": [Function],
  "resolvePages": [Function],
  "resolveSiteUrl": [Function],
  "serialize": [Function],
}
`;
